tests/data/github/repos.py

```diff
@@ -30,6 +30,7 @@
             'login': 'example_org',
             '__typename': 'Organization',
         },
+        'requirements': {'text': 'cartography\nhttplib2>=0.7.0\njinja2\nlxml\n'},
```

Comment:
I would recommend having some `==` in here and less unpinned deps as requirements.txt files should have fully pinned deps (as the output from piptools-compile). Still reasonable to have test cases for unpinned/loosely bound since (in OSS/industry) not everyone pins everything in requirements.txt and we should degrade gracefully.

Reply:
Good idea, I'll add these test cases.

Reply:
Added, see the test case below for SampleRepo2.